### PR TITLE
[Model] Support quantization of PixtralHFTransformer for PixtralHF

### PR DESCRIPTION
**Pull request by mgoin**
Allows us to support quantization of the vision encoder for PixtralHF, where before we could only quantize the language model. Also converts the raw torch nn.Linear modules to proper merged vLLM Linear modules

Tested with 
* `neuralmagic/pixtral-12b-FP8-dynamic` (only `MistralModel` quantized)
* `nm-testing/pixtral-12b-FP8-dynamic-all` (`MistralModel` and `PixtralHFTransformer` quantized, just not `lm_head`)

### CODE REVIEW

File: vllm/model_executor/models/pixtral.py [hunk 1]
```diff
@@ -788,20 +791,25 @@ def __init__(
...
+                                           prefix=f"{prefix}.down_proj")
         self.act = get_act_fn(config.hidden_act)
 
     def forward(self, x: torch.Tensor) -> torch.Tensor:
-        return self.down_proj(self.act(self.gate_proj(x)) * self.up_proj(x))
+        gate_up, _ = self.gate_up_proj(x)
+        d = gate_up.shape[-1] // 2
+        x = self.act(gate_up[..., :d]) * gate_up[..., d:]
+        x, _ = self.down_proj(x)
+        return x
```
Comment (jeejeelee) [inline vllm/model_executor/models/pixtral.py:812]: Can we use the `act_and_mul` kernel?

Comment (mgoin) [reply vllm/model_executor/models/pixtral.py:812]: There are some pixtral checkpoints out there that use gelu, so I didn't want to make those stop working by only supporting SiluAndMul

See some models here:
* https://huggingface.co/SeanScripts/pixtral-12b-nf4/blob/ce4c79e0a2d82b2f8f4ff66a65c95cd4564ef492/config.json#L128
* https://huggingface.co/Ertugrul/Pixtral-12B-Captioner-Relaxed/blob/768ab0dc91f78088f523167f02158877a5f0b240/config.json#L113

Comment (LucasWilkinson) [reply vllm/model_executor/models/pixtral.py:812]: Maybe not the PR but there is a `GeluAndMul` module,

why not make a `get_act_and_mul_fn` to replace `get_act_fn` in the init?

Comment (mgoin) [reply vllm/model_executor/models/pixtral.py:812]: This is a fair point, I can make this

**Review by ywang96 — APPROVED** [review]
LGTM! Not related to this PR in particular but does it make sense to use `get_act_and_mul_fn` and `act_and_mul` everywhere?

**Comment by mgoin** [comment]
Yeah this was my thinking @ywang96, with `get_act_and_mul_fn` now we should be able to replace most hardcoded SiluAndMul usages. First I'm going to look to see if we need this `ScaledActivation` case anymore

**Pull request merged**